conans/client/tools/oss.py

```diff
@@ -489,17 +491,20 @@ def get_gnu_triplet(os_, arch, compiler=None):
 def check_output(cmd, folder=None, return_code=False, stderr=None):
     tmp_file = tempfile.mktemp()
     try:
+        # We don't want stderr to print warnings that will mess the pristine outputs
         stderr = stderr or PIPE
         cmd = cmd if isinstance(cmd, six.string_types) else subprocess.list2cmdline(cmd)
         process = subprocess.Popen("{} > {}".format(cmd, tmp_file), shell=True,
                                    stderr=stderr, cwd=folder)
-        process.communicate()
+
+        _, stderr = process.communicate()
 
         if return_code:
             return process.returncode
 
         if process.returncode:
-            raise CalledProcessError(process.returncode, cmd)
+            # Only in case of error, we print also the stderr to know what happened
+            raise CalledProcessErrorWithStderr(process.returncode, cmd, stderr=stderr)
```

Comment:
https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError

> Changed in version 3.5: stdout and stderr attributes added

Will crash in other python versions and in 2.7

Reply:
Yes, fixed.